core/app/models/spree/line_item.rb

```diff
@@ -138,49 +114,68 @@ def options=(options={})
       opts = options.dup # we will be deleting from the hash, so leave the caller's copy intact
 
       currency = opts.delete(:currency) || order.try(:currency)
```

Comment:
Nope, `currency` here is a local variable and is not passed into the `LineItem` instance.